File: android/src/main/java/com/mrousavy/nearby/GoogleNearbyMessagesPackage.java
package com.mrousavy.nearby;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;

import com.facebook.react.ReactPackage;
import com.facebook.react.bridge.NativeModule;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.uimanager.ViewManager;
import com.facebook.react.bridge.JavaScriptModule;

public class GoogleNearbyMessagesPackage implements ReactPackage {
    @Override
    public List<NativeModule> createNativeModules(ReactApplicationContext reactContext) {
        return Arrays.<NativeModule>asList(new GoogleNearbyMessagesModule(reactContext));
    }

    public List<Class<? extends JavaScriptModule>> createJSModules() {
        // deprecated in react native 0.47
        return null;
    }

    @Override
    public List<ViewManager> createViewManagers(ReactApplicationContext reactContext) {
        return Collections.emptyList();
    }
}

File: android/src/main/java/com/mrousavy/nearby/GoogleNearbyMessagesPackage.kt
package com.mrousavy.nearby

import com.facebook.react.ReactPackage
import com.facebook.react.bridge.JavaScriptModule
import com.facebook.react.bridge.NativeModule
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.uimanager.ViewManager
import java.util.*

class GoogleNearbyMessagesPackage : ReactPackage {
    override fun createNativeModules(reactContext: ReactApplicationContext): List<NativeModule> {
        return Arrays.asList<NativeModule>(GoogleNearbyMessagesModule(reactContext))
    }

    fun createJSModules(): List<Class<out JavaScriptModule?>>? {
        // deprecated in react native 0.47
        return null
    }

    override fun createViewManagers(reactContext: ReactApplicationContext): List<ViewManager<*, *>> {
        return emptyList()
    }
}